src/Model.jl

```diff
@@ -175,6 +175,26 @@ function Model(lattice::AbstractMatrix{<:Quantity}, atoms::Vector{<:Element},
     Model(austrip.(lattice), atoms, positions; kwargs...)
 end
 
+"""
+    Model(model, lattice, positions)
+    Model(model, lattice)
+    Model(model, positions)
+
+Construct an identical model to `model` with the option to change `lattice` or `positions`
+(useful for geometry or lattice optimisations).
+"""
+function Model(model::Model{T}, lattice::AbstractMatrix{U},
```

Comment:
In general, not a huge fan of this type of constructor (ie why change one field instead of another?). How about `Model(model; lattice, positions)`? (which we can then extend to other things if needed)

Reply:
The advantage of not taking kwargs is to do slightly different think depending on what is passed. E.g. for changing kcoords we certainly do not want to change the FFT size, but for updating positions or lattice we certainly do. If we move all to a kwargs-based update constructor it's not so clear to do.

Reply:
pwbasis is something else, but Model is a relatively simple container with not much going on in the constructor, so I think it definitely makes sense to have a Model(model; kwargs). I don't get your `Model(model)` argument?